### PR TITLE
Removing Await.results: 1 down 6 to go.

### PR DESCRIPTION
**Pull request by mcovarr**
Not too happy that I had to change the way the FSM test worked, but the WorkflowActor seemed to be running fine.

### CODE REVIEW

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 1]
```diff
@@ -33,6 +31,25 @@ object WorkflowActor {
   case class CallCompleted(call: OutputKey, callOutputs: CallOutputs, returnCode: Int) extends WorkflowActorMessage
   case class CallFailed(call: OutputKey, returnCode: Option[Int], failure: String) extends WorkflowActorMessage
   case object Terminate extends WorkflowActorMessage
+  case class ExecutionStoreCreated(startMode: StartMode) extends WorkflowActorMessage
+  case class AsyncFailure(t: Throwable) extends WorkflowActorMessage
+
+  sealed trait StartMode {
+    def runInitialization(actor: WorkflowActor): Future[Unit]
+  }
+  case object Start extends WorkflowActorMessage with StartMode {
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: \n

File: src/main/scala/cromwell/engine/workflow/WorkflowActor.scala [hunk 2]
```diff
@@ -33,6 +31,25 @@ object WorkflowActor {
...
+    override def runInitialization(actor: WorkflowActor): Future[Unit] = {
+      // This only does the initialization for a newly created workflow.  For a restarted workflow we should be able
+      // to assume the adjusted symbols already exist in the DB, but is it safe to assume the staged files are in place?
+      actor.initializeWorkflow match {
+        case Success(inputs) => actor.createWorkflow(inputs)
+        case Failure(ex) => Future.failed(ex)
+      }
+    }
+  }
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/workflow/WorkflowActor.scala]: \n

**Comment by geoffjentry** [comment]
:+1:

**Comment by mcovarr** [comment]
Please don't merge this PR yet.  I believe the changes I've made are okay but tests are failing pretty consistently for reasons that aren't clear yet.

**Comment by mcovarr** [comment]
OK it looks like the tests are passing consistently now, so unless you see breakage that you think could be related to these changes, this is fair game for review and merge.

**Comment by geoffjentry** [comment]
This seems like a good candidate to expose to some posthumous you know whating, at least to the extent that it's possible currently.

**Comment by mcovarr** [comment]
Given the current Tyburn bent toward JES workflows with GCS inputs and Docker images, the standard battery isn't really compatible with stress testing these changes.  But I can make some three-step and scatter WDLs and do some custom posthumous execution.

**Comment by mcovarr** [comment]
I've subjected this to 100 iterations of 10-wide three-step + scatter workflows and it seems fine, another thumbs up and I think this is mergeable.

**Comment by Horneth** [comment]
:+1:

**Pull request merged**